### PR TITLE
ResponsiveBox - Applying screenSize css class

### PR DESCRIPTION
**Pull request by novsstation**
Applying screenSize css class

### CODE REVIEW

File: js/renovation/ui/responsive_box/__tests__/responsive_box.test.tsx [hunk 1]
```diff
@@ -1,20 +1,70 @@
...
+    ]).describe('Config: %o', ({ clientWidth, expectedClass }) => {
+      it('Check has valid screen size. Default implementation. Has window', () => {
+        const defaultImplementation = domAdapter.getDocumentElement;
+        domAdapter.getDocumentElement = () => ({ clientWidth });
+
+        const props = new ResponsiveBoxProps();
+        const responsiveBox = mount<ResponsiveBox>(<ResponsiveBox {...props} />);
+        const { classList } = responsiveBox.getDOMNode();
+
+        expect(classList.contains('dx-responsivebox')).toEqual(true);
```
Comment (IgnatovDan) [inline js/renovation/ui/responsive_box/__tests__/responsive_box.test.tsx]: Can be tested in a separate test without "each(["
Of course it should be in each config but our current testing religion suggests to make a separate test.

File: js/renovation/ui/responsive_box/responsive_box.tsx [hunk 2]
```diff
@@ -5,8 +5,31 @@ import { Widget } from '../common/widget';
...
+import { convertToScreenSizeQualifier } from './screen_utils';
+
+const HD_SCREEN_WIDTH = 1920;
+const RESPONSIVE_BOX_CLASS = 'dx-responsivebox';
+const SCREEN_SIZE_CLASS_PREFIX = `${RESPONSIVE_BOX_CLASS}-screen-`;
+
+export const viewFunction = (viewModel: ResponsiveBox): JSX.Element => {
+  const getCurrentScreenSizeQualifier = (): ScreenSizeQualifier => {
+    const screenWidth = hasWindow()
+      ? domAdapter.getDocumentElement().clientWidth
```
Comment (IgnatovDan) [inline js/renovation/ui/responsive_box/responsive_box.tsx:20]: There was "return hasWindow() ? $(window).width() : HD_SCREEN_WIDTH;"
But "domAdapter.getDocumentElement().clientWidth" is used now.
Why?

https://js.devexpress.com/Documentation/21_1/ApiReference/UI_Components/dxResponsiveBox/Configuration/#screenByWidth
"accepts the screen's width" - this description seems to be outdated? Our code doesn't pass "screen's width" to this method.

hg/mobile/rev/1a0534a45211

**Review by IgnatovDan — APPROVED** [review]

File: js/renovation/ui/responsive_box/responsive_box.tsx (hunk 2 above)
Comment (novsstation) [reply js/renovation/ui/responsive_box/responsive_box.tsx:20]: > But "domAdapter.getDocumentElement().clientWidth" is used now. Why

It is analog of the computing of the window width except scrollbar width.
Inner width and outerWidth of the window object returns width without checking of the visibility of the scrollbar
But clientWidth of the document check it:

image

File: js/renovation/ui/responsive_box/__tests__/responsive_box.test.tsx (hunk 1 above)
Comment (novsstation) [reply js/renovation/ui/responsive_box/__tests__/responsive_box.test.tsx]: Yep, fixed :)

**Review by EugeniyKiyashko — APPROVED** [review]

**Pull request merged**